src/cogent3/core/tree.py

```diff
@@ -2179,6 +2179,39 @@ def set_max_tip_tip_distance(self):
                 n.MaxDistTips = [tip_a, tip_b]
 
 
+def split_name_and_support(name_field: str | None) -> tuple[str | None, float | None]:
+    """Handle cases in the Newick format where an internal node name field
+    contains a name or/and support value, like 'edge.98/100'.
+    """
+    # handle the case where the name field is an empty string or None
+    if not name_field:
+        return None, None
+
+    # split name and support by forward slash
+    parts = name_field.split("/")
+
+    if len(parts) == 2:
+        name = parts[0]
+        # the support value should be either a float or an int
+        try:
+            support = float(parts[1])
+        except ValueError as e:
+            raise ValueError(
+                f"Support value at node: {name} should be int or float."
+            ) from e
+        return name, support
+    elif len(parts) == 1:
```

Comment:
This code can be made much simpler with the following.
```python
name, *support = name_field.split("/")
name = name or None
support = support or None
try:
    support = float(support)
except (ValueError, TypeError) as e:
    raise ValueError(
                f"Support value at node: {name!r} should be int or float not {support!r}."
            ) from e
```

that covers all your cases I believe